[Security] 8.18.4 release notes (#6912)
elastic/security-docs: docs/release-notes/8.18.asciidoc

[[release-notes-header-8.18.0]]
== 8.18

[discrete]
[[release-notes-8.18.4]]
=== 8.18.4

[discrete]
[[enhancements-8.18.4]]
==== Enhancements

> * Improves logging of fatal exceptions.

Author: ```suggestion
* Improves logging of fatal exceptions in {elastic-endpoint}.
```

[discrete]
[[bug-fixes-8.18.4]]
==== Fixes
* Fixes domain validation in case observables ({kibana-pull}225901[#225901]).
* Updates a placeholder and validation message in the **Related Integrations** section of the rule upgrade flyout ({kibana-pull}225775[#225775]).

> * Excludes {{ml}} rules from installation and upgrade checks for users with Basic or Essentials licenses ({kibana-pull}224676[#224676]).

Author: ```suggestion
* Excludes {ml} rules from installation and upgrade checks for users with Basic or Essentials licenses ({kibana-pull}224676[#224676]).
```

[[release-notes-header-8.18.0]]
== 8.18

[discrete]
[[release-notes-8.18.4]]
=== 8.18.4

[discrete]
[[enhancements-8.18.4]]
==== Enhancements

> * Improves logging of fatal exceptions in {elastic-endpoint}.

Reviewer: ```suggestion
* Improves logging of fatal exceptions in {elastic-defend}.
* {elastic-defend} users can now control the maximum file size for malware protection via the `advanced.malware.max_file_size_bytes` advanced policy option.
```

[discrete]
[[bug-fixes-8.18.4]]
==== Fixes
* Fixes differences between risk scoring preview and persisted risk scores ({kibana-pull}226456[#226456]).
* Fixes domain validation in case observables ({kibana-pull}225901[#225901]).
* Updates a placeholder and validation message in the **Related Integrations** section of the rule upgrade flyout ({kibana-pull}225775[#225775]).
* Excludes {ml} rules from installation and upgrade checks for users with Basic or Essentials licenses ({kibana-pull}224676[#224676]).
* Allows using days as a time unit in rule schedules, fixing an issue where durations normalized to days were incorrectly displayed as 0 seconds ({kibana-pull}224083[#224083]).
* Strips `originId` from connectors before rule import to ensure correct ID regeneration and prevent errors when migrating connector references on rules ({kibana-pull}223454[#223454]).
* Refactors Timeline styling for improved consistency with design updates ({kibana-pull}222438[#222438]).

> * Fixes a bug where the **Rules**, **Alerts**, and **Fleet** pages would stall in air-gapped environments by ensuring API requests are sent even when offline ({kibana-pull}220510[#220510]).

Reviewer: ```suggestion
* Fixes a bug where the **Rules**, **Alerts**, and **Fleet** pages would stall in air-gapped environments by ensuring API requests are sent even when offline ({kibana-pull}220510[#220510]).
* Fixes an issue in {elastic-defend} that may result in bugchecks (BSODs) on Windows systems with a very high volume of network connections. This issue has only been observed on Windows Server.
```

> * Fixes a bug where the **Rules**, **Alerts**, and **Fleet** pages would stall in air-gapped environments by ensuring API requests are sent even when offline ({kibana-pull}220510[#220510]).

Reviewer 2: ```suggestion
* Fixes a bug where the **Rules**, **Alerts**, and **Fleet** pages would stall in air-gapped environments ({kibana-pull}220510[#220510]).
```

I'd say we can omit the implementation details. If someone is interested, they can find the specifics in the PR.

[discrete]
[[bug-fixes-8.18.4]]
==== Fixes
* Fixes differences between risk scoring preview and persisted risk scores ({kibana-pull}226456[#226456]).
* Fixes domain validation in case observables ({kibana-pull}225901[#225901]).
* Updates a placeholder and validation message in the **Related Integrations** section of the rule upgrade flyout ({kibana-pull}225775[#225775]).
* Excludes {ml} rules from installation and upgrade checks for users with Basic or Essentials licenses ({kibana-pull}224676[#224676]).
* Allows using days as a time unit in rule schedules, fixing an issue where durations normalized to days were incorrectly displayed as 0 seconds ({kibana-pull}224083[#224083]).
* Strips `originId` from connectors before rule import to ensure correct ID regeneration and prevent errors when migrating connector references on rules ({kibana-pull}223454[#223454]).
* Refactors Timeline styling for improved consistency with design updates ({kibana-pull}222438[#222438]).
* Fixes a bug where the **Rules**, **Alerts**, and **Fleet** pages would stall in air-gapped environments ({kibana-pull}220510[#220510]).

> * Fixes an issue in {elastic-defend} that may result in bugchecks (BSODs) on Windows systems with a very high volume of network connections. This issue has only been observed on Windows Server.

Reviewer 3: Hey @natasha-moore-elastic,

We just got merged https://github.com/elastic/kibana/pull/227859 and https://github.com/elastic/kibana/pull/227803. I see these two aren't in the list.

@nikitaindik and @dplumlee are there any other missing bug fixes you backported to `8.18.4`?

Author: Hi @maximpn – https://github.com/elastic/kibana/pull/227355 doesn't have an `v8.18.4` label and I also don't see an 8.18 backport in the PR, so just double-checking it will indeed go into 8.18.4?

I'll add https://github.com/elastic/kibana/pull/227803 to the list of fixes 👍

Reviewer 3: @natasha-moore-elastic, sorry I accidentally copied a wrong link. I meant https://github.com/elastic/kibana/pull/227859.

Reviewer 4: > so just double-checking it will indeed go into 8.18.4?

@natasha-moore-elastic No, https://github.com/elastic/kibana/pull/227355 a fix for the feature that's not released yet. Hence the `release_note:skip` label. No need to mention it here.

Feel free to add a note for the https://github.com/elastic/kibana/pull/227803 one.

Author: Added https://github.com/elastic/kibana/pull/227859 and https://github.com/elastic/kibana/pull/227803 in https://github.com/elastic/security-docs/pull/6912/commits/ae675323efe0931407d31ca7b19e7ace1a5e8d27
